source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,214 @@
+   as the procedure is complete. Otherwise propagate the operation failure
+   error(*).
+6. Connect to the server E and execute the `replSetStepDown
+   <https://docs.mongodb.com/manual/reference/command/replSetStepDown/>`_
+   admin command as follows:
+   ``{replSetStepDown: 4, secondaryCatchUpPeriodSecs: 2}``.
+7. Connect directly to the server N and execute the following admin command:
+   ``{replSetStepUp: 1}``. If this command fails with an operation failure
+   error with code 125 ("Election failed"), repeat this step 7.
+8. Set the status of all servers in the topology to Unknown.
+9. Perform server selection to obtain the current primary.
+10. If the current primary is not N, go to step 7 and step up the server N
+    again.
+
+(*) The cluster may have held an election after the replica set was reconfigured
+in step 4, making the server N the current primary. If this happens,
+``replSetFreeze`` command will fail, but the overall goal of having the
+server N as the primary has been achieved.
+
+As the above procedure contains a potentially infinite loop, the driver MAY
+impose a time limit and fail if the server N has not become a primary in the
+allotted time. This time limit SHOULD be at least 10 seconds.
+
+Stepping Down A Server
+``````````````````````
+
+If the driver does not care which server becomes the new primary, as would be
+the case in the getMore test for example, the driver should perform the
+following procedure:
+
+- Obtain a list of servers in the cluster.
+- Randomly or otherwise choose a server to be the new primary, other than the
+  current primary.
+- Follow the above server election procedure to elect the chosen server as
+  the new primary.
+
+Tests
+-----
+
+The driver should implement the following tests:
+
+getMore Iteration
+`````````````````
+
+This test requires server version 4.2 or higher.
+
+Perform the following operations:
+
+- Insert 5 documents into a collection.
+- Start a find operation on the collection with a batch size of 2, and
+  retrieve the first batch of results.
+- Step down the primary, and verify that another server has been elected as
+  a primary.
+- Using the same connection that was used for the initial find, continue
+  iterating the collection by issuing a getMore operation.
+- Verify that the next batch of results is returned successfully.
+
+Drivers that do not ordinarily allow applications to execute operations
+on specific connections may need to perform the getMore iteration test on
+a lower level than the public API exposed to applications.
+In this case the driver SHOULD verify that both initial find and the
+subsequent getMore are dispatched on the same connection, for example by
+comparing socket handles used for the two operations.
+
+Acknowledged Write
+``````````````````
+
+This test requires server version 4.2 or higher.
+
+Perform the following operations:
+
+- Obtain a connection to the current primary.
```

Comment:
Seems like this also needs the "may need lower level than public API" comment as I imagine many drivers don't easily expose specific connections for operations.

I wonder if these tests are going to need or could more easily be done some configuration specific to the test, e.g.:
* turn off retryable writes
* set connection pool to size 1

Reply:
Setting connection pool size to 1 does not necessarily preclude a driver from using a different connection for each operation.

Reply:
It doesn't preclude it, but statistically, it's likely.